src/osd/PG.cc

```diff
@@ -4555,8 +4560,29 @@ int PG::build_scrub_map_chunk(
   // finish
   dout(20) << __func__ << " finishing" << dendl;
   assert(pos.done());
-  _scan_snaps(map);
   _repair_oinfo_oid(map);
+  if (!is_primary()) {
+    ScrubMap for_meta_scrub;
+    scrubber.cleaned_meta_map.insert(map);
+    if (end.is_max() || scrubber.cleaned_meta_map.objects.empty()) {
+      scrubber.cleaned_meta_map.swap(for_meta_scrub);
+    } else {
+      auto iter = scrubber.cleaned_meta_map.objects.end();
+      --iter; // not empty, see if clause
+      auto begin = scrubber.cleaned_meta_map.objects.begin();
+      while (iter != begin) {
+        auto next = iter--;
+        if (next->first.get_head() != iter->first.get_head()) {
+	  ++iter;
+	  break;
+        }
+      }
+      for_meta_scrub.objects.insert(begin, iter);
+      scrubber.cleaned_meta_map.objects.erase(begin, iter);
```

Comment:
Can we move this into a helper so that we aren't duplicating the fragment from scrub_compare_maps?  Something like Scrubber::clean_meta_map(map, &for_meta_scrub), where the helper handles the insert/swap/erase dance?